python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -0,0 +1,372 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = (
+    "AssembleCellCoaddTask",
+    "AssembleCellCoaddConfig",
+    "ConvertMultipleCellCoaddToExposureTask",
+)
+
+
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import numpy as np
+from lsst.cell_coadds import (
+    CellIdentifiers,
+    CoaddUnits,
+    CommonComponents,
+    GridContainer,
+    MultipleCellCoadd,
+    OwnedImagePlanes,
+    PatchIdentifiers,
+    SingleCellCoadd,
+    UniformGrid,
+)
+from lsst.meas.algorithms import AccumulatorMeanStack, CoaddPsf, CoaddPsfConfig
+from lsst.pex.config import ConfigField, ConfigurableField, Field, ListField, RangeField
+from lsst.pipe.base import NoWorkFound, PipelineTask, PipelineTaskConfig, PipelineTaskConnections, Struct
+from lsst.pipe.base.connectionTypes import Input, Output
+from lsst.pipe.tasks.coaddBase import makeSkyInfo
+from lsst.pipe.tasks.coaddInputRecorder import CoaddInputRecorderTask
+from lsst.pipe.tasks.interpImage import InterpImageTask
+from lsst.pipe.tasks.scaleZeroPoint import ScaleZeroPointTask
+from lsst.skymap import BaseSkyMap
+
+
+class AssembleCellCoaddConnections(
+    PipelineTaskConnections,
+    dimensions=("tract", "patch", "band", "skymap"),
+    defaultTemplates={"inputWarpName": "deep", "outputCoaddSuffix": "Cell"},
+):
+    inputWarps = Input(
+        doc="Input warps",
+        name="{inputWarpName}Coadd_directWarp",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "visit", "instrument"),
+        deferLoad=True,
+        multiple=True,
+    )
+
+    skyMap = Input(
+        doc="Input definition of geometry/bbox and projection/wcs",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    multipleCellCoadd = Output(
+        doc="Output multiple cell coadd",
+        name="{inputWarpName}Coadd{outputCoaddSuffix}",
+        storageClass="MultipleCellCoadd",
+        dimensions=("tract", "patch", "band", "skymap"),
+    )
+
+
+class AssembleCellCoaddConfig(PipelineTaskConfig, pipelineConnections=AssembleCellCoaddConnections):
+    doInterp = Field[bool](doc="Interpolate over NaN pixels?", default=False)
+    interpImage = ConfigurableField(
+        target=InterpImageTask,
+        doc="Task to interpolate images (and extrapolate) over NaN pixels",
+    )
+    scaleZeroPoint = ConfigurableField(
+        target=ScaleZeroPointTask,
+        doc="Task to scale images to a common zero point",
+    )
+    badMaskPlanes = ListField[str](
+        doc="Mask planes that, if set, the corresponding pixel would not be included in the coadd.",
+        default=("BAD", "EDGE", "NO_DATA", "SAT"),
+    )
+    calcErrorFromInputVariance = Field[bool](
+        doc="Calculate coadd variance from input variance by stacking "
+        "statistic. Passed to "
+        "StatisticsControl.setCalcErrorFromInputVariance()",
```

Comment:
Is the alternative to use a single weight for the whole warp?

Reply:
I think so.  We'd prefer for it to be a per-visit or per-{visit+detector} weight, so that it's the same across different warps, but we can't do that without at least moving the scaling to `makeWarp`, and that's not a job for this ticket.